server/src/main/java/org/apache/iotdb/db/qp/strategy/LogicalGenerator.java

```diff
@@ -1411,6 +1418,16 @@ private Long parseDuration(String durationStr) {
           i++;
           unit += durationStr.charAt(i);
         }
+        if (unit.toLowerCase().equals("mo")) {
+          //interval is by month, sliding step by default equals to interval
+          if (!isParsingSlidingStep) {
+            queryOp.setIntervalByMonth(true);
+          }
+          queryOp.setSlidingStepByMonth(true);
+        } else if (isParsingSlidingStep) {
+          //parsing sliding step value, and unit is not by month
+          queryOp.setSlidingStepByMonth(false);
+        }
```

Comment:
Hi, thank you for your review. 
In parseDuration clause, it converts the interval and sliding step in milliseconds, also determines the unit for natural month aggregation. If I put the parsing "mo" in enterGroupByTimeClause(). i think it will be redoing the same thing.